tests/test_config_parsing/summary_generator.py

```diff
@@ -0,0 +1,304 @@
+    section 11.2.1 in opm flow reference manual 2022-10.
+    """
+    first_character = draw(st.sampled_from("ABFGRWCS"))
+    if first_character == "A":
+        second_character = draw(st.sampled_from("ALN"))
+        third_character = draw(st.sampled_from("QL"))
+        fourth_character = draw(st.sampled_from("RT"))
+        return first_character + second_character + third_character + fourth_character
+
+    kind = draw(st.sampled_from([1, 2, 3, 4]))
+    if kind == 1:
+        return draw(
+            st.sampled_from(
+                ["BAPI", "BOSAT", "BPR", "FAQR", "FPR", "FWCT", "WBHP", "WWCT"]
+            )
+        )
+    elif kind == 2:
+        direction = draw(st.sampled_from("IJK"))
+        return (
+            draw(st.sampled_from(["FLOO", "VELG", "VELO", "FLOW", "VELW"])) + direction
+        )
+    elif kind == 3:
+        dimension = draw(st.sampled_from("XYZRT"))
+        direction = draw(st.sampled_from(" -"))
+        return draw(st.sampled_from(["GKR", "OKR", "WKR"])) + dimension + direction
+    else:
+        second_character = draw(st.sampled_from("OWGVLPT"))
+        third_character = draw(st.sampled_from("PIF"))
+        fourth_character = draw(st.sampled_from("RT"))
+        return first_character + second_character + third_character + fourth_character
+
+
+unit_names = st.sampled_from(
+    ["SM3/DAY", "BARSA", "SM3/SM3", "FRACTION", "DAYS", "YEARS", "SM3", "SECONDS"]
+)
+
+names = st.text(
+    min_size=8, max_size=8, alphabet=st.characters(min_codepoint=65, max_codepoint=90)
+)
+
+
+@unique
+class UnitSystem(Enum):
+    METRIC = 1
+    FIELD = 2
+    LAB = 3
+
+    def to_ecl(self):
+        return self.value
+
+
+@unique
+class Simulator(Enum):
+    ECLIPSE_100 = 100
+    ECLIPSE_300 = 300
+    ECLIPSE_300_THERMAL = 500
+    INTERSECT = 700
+    FRONTSIM = 800
+
+    def to_ecl(self):
+        return self.value
+
+
+@dataclass
+class SmspecIntehead(GrdeclKeyword):
+    unit: UnitSystem
+    simulator: Simulator
+
+
+@dataclass
+class Date:
+    day: conint(ge=1, le=31)  # type: ignore
+    month: conint(ge=1, le=12)  # type: ignore
+    year: conint(gt=1901, lt=2038)  # type: ignore
+    hour: conint(ge=0, lt=24)  # type: ignore
+    minutes: conint(ge=0, lt=60)  # type: ignore
+    micro_seconds: conint(ge=0, lt=60000000)  # type: ignore
+
+    def to_ecl(self):
+        return astuple(self)
+
+
+@dataclass
+class Smspec:
+    intehead: SmspecIntehead
+    restart: str
+    num_keywords: PositiveInt
+    nx: PositiveInt
+    ny: PositiveInt
+    nz: PositiveInt
+    restarted_from_step: PositiveInt
+    keywords: List[str]
+    well_names: List[str]
+    region_numbers: List[str]
+    units: List[str]
+    start_date: Date
+
+    def to_ecl(self) -> List[Tuple[str, Any]]:
+        restart = self.restart.ljust(72, " ")
+        return [
+            ("INTEHEAD", self.intehead.to_ecl()),
+            ("RESTART ", [restart[i * 8 : i * 8 + 8] for i in range(9)]),
```

Comment:
Please comment why these magic numbers actually work, or at least where the numbers origin from. I vaguely remember you commented on this during a coding session.

Reply:
Done.